File: backend/StudyFlow.Domain/Entities/Quiz.cs
namespace StudyFlow.Domain.Entities;

public class Quiz
{
    public int Id { get; set; }
    public string Name { get; set; } = default!;
    public DateTime CreatedAt { get; set; }

    public int TopicId { get; set; }

    public List<Question> Questions { get; set; } = [];
}

File: frontend/lib/models/quiz.dart
class Quiz {
  const Quiz({
    required this.id,
    required this.topicId,
    required this.name,
    required this.createdAt,
    this.updatedAt,
  });

  final String id;
  final String topicId;
  final String name;
  final DateTime createdAt;
  final DateTime? updatedAt;
}
